### PR TITLE
adding arm64 more recent version of node

### PR DESCRIPTION
**Pull request by bjornmonnens**
For my local RPI4 ubuntu based docker I needed a more recent version of node. Installed the the plugin build with this branch and now I am able to use the Node 15.8.0. On the main plugin this failed as it only goes up to 12.

### CODE REVIEW

**Comment by nfalco79** [comment]
Could you provide additional test case in InstallerPathResolversTest ?

File: src/main/java/jenkins/plugins/nodejs/tools/pathresolvers/LatestInstallerPathResolver.java [hunk 1]
```diff
@@ -114,10 +114,17 @@ public String resolvePathFor(String version, Platform platform, CPU cpu) {
             arch = "x64";
             break;
         case arm64:
+            if (nodeVersion.compareTo(new NodeJSVersion(15, 8, 1)) >= 0 || nodeVersion.compareTo(new NodeJSVersion(8, 6, 0)) == 0 || nodeVersion.compareTo(new NodeJSVersion(4, 0, 0)) < 0) {
```
Comment (nfalco79) [inline src/main/java/jenkins/plugins/nodejs/tools/pathresolvers/LatestInstallerPathResolver.java:117]: It's not clear why you ensure the version is lower than 15.8.1 (actually is not released yet) ? Do you expect arm64 will no more supported by nodejs?

**Review by nfalco79 — CHANGES_REQUESTED** [review]

File: src/main/java/jenkins/plugins/nodejs/tools/pathresolvers/LatestInstallerPathResolver.java (hunk 1 above)
Comment (bjornmonnens) [reply src/main/java/jenkins/plugins/nodejs/tools/pathresolvers/LatestInstallerPathResolver.java]: I removed the upper bound of 15.8.1, so should automatically allow to download newer versions.

**Comment by bjornmonnens** [comment]
I've updates the test suite to contain the 15.8.0 entry, test should be green

File: src/main/java/jenkins/plugins/nodejs/tools/pathresolvers/LatestInstallerPathResolver.java [hunk 2]
```diff
@@ -105,7 +105,10 @@ public String resolvePathFor(String version, Platform platform, CPU cpu) {
             break;
         case amd64:
             if (platform == Platform.SUNOS && //
-                    (new NodeJSVersionRange("[7, 7.5)").includes(nodeVersion) || nodeVersion.compareTo(new NodeJSVersion(0, 12, 18)) == 0)) {
+                    (   new NodeJSVersionRange("[7, 7.5)").includes(nodeVersion) ||
```
Comment (nfalco79) [inline src/main/java/jenkins/plugins/nodejs/tools/pathresolvers/LatestInstallerPathResolver.java:108]: Please revert changes not related to this PR

File: src/main/java/jenkins/plugins/nodejs/tools/pathresolvers/LatestInstallerPathResolver.java [hunk 3]
```diff
@@ -114,10 +117,17 @@ public String resolvePathFor(String version, Platform platform, CPU cpu) {
             arch = "x64";
             break;
         case arm64:
+            if (nodeVersion.compareTo(new NodeJSVersion(8, 6, 0)) == 0 || nodeVersion.compareTo(new NodeJSVersion(4, 0, 0)) < 0) {
```
Comment (nfalco79) [inline src/main/java/jenkins/plugins/nodejs/tools/pathresolvers/LatestInstallerPathResolver.java:120]: Why 8.6.0 is excluded from download?

File: src/test/resources/updates/jenkins.plugins.nodejs.tools.NodeJSInstaller.json [hunk 4]
```diff
@@ -1,2277 +1,2284 @@
-{"list": [
```
Comment (nfalco79) [inline src/test/resources/updates/jenkins.plugins.nodejs.tools.NodeJSInstaller.json:1]: Too many changes due reformatting, i'm not able to review changes

**Review by nfalco79 — CHANGES_REQUESTED** [review]

File: src/main/java/jenkins/plugins/nodejs/tools/pathresolvers/LatestInstallerPathResolver.java (hunk 2 above)
Comment (bjornmonnens) [reply src/main/java/jenkins/plugins/nodejs/tools/pathresolvers/LatestInstallerPathResolver.java]: Changes reverted (note that that version is not available as download)

File: src/main/java/jenkins/plugins/nodejs/tools/pathresolvers/LatestInstallerPathResolver.java (hunk 3 above)
Comment (bjornmonnens) [reply src/main/java/jenkins/plugins/nodejs/tools/pathresolvers/LatestInstallerPathResolver.java:117]: This was already part of the initial code base
https://github.com/jenkinsci/nodejs-plugin/blob/master/src/main/java/jenkins/plugins/nodejs/tools/pathresolvers/LatestInstallerPathResolver.java#L118

File: src/test/resources/updates/jenkins.plugins.nodejs.tools.NodeJSInstaller.json (hunk 4 above)
Comment (bjornmonnens) [reply src/test/resources/updates/jenkins.plugins.nodejs.tools.NodeJSInstaller.json:1]: reverted the json file to the previous version and added the entries.
Might be good to reformat the file to correct the indentation

File: src/main/java/jenkins/plugins/nodejs/tools/pathresolvers/LatestInstallerPathResolver.java (hunk 3 above)
Comment (nfalco79) [reply src/main/java/jenkins/plugins/nodejs/tools/pathresolvers/LatestInstallerPathResolver.java:117]: Yes but the baseline was wrong. For arm64
https://nodejs.org/dist/v8.6.0/node-v8.6.0-linux-arm64.tar.gz

Comment (bjornmonnens) [reply src/main/java/jenkins/plugins/nodejs/tools/pathresolvers/LatestInstallerPathResolver.java]: I removed the baseline.
Kept the case arm64 and armv7l as seperate cases (although the body is the same)

File: src/test/resources/updates/jenkins.plugins.nodejs.tools.NodeJSInstaller.json (hunk 4 above)
Comment (bjornmonnens) [reply src/test/resources/updates/jenkins.plugins.nodejs.tools.NodeJSInstaller.json:1]: After couple of commits, the review should now be clear. Seems the idea does automatic indentation upon paste.

Comment (nfalco79) [reply src/test/resources/updates/jenkins.plugins.nodejs.tools.NodeJSInstaller.json:1]: You need just to download the updated file from http://updates.jenkins-ci.org/updates/hudson.plugins.nodejs.tools.NodeJSInstaller.json and remove the non JSON part `downloadService.post('hudson.plugins.nodejs.tools.NodeJSInstaller',` from begin and `)` from end without apply any formatting

**Review by nfalco79 — CHANGES_REQUESTED** [review]
The other parts seems to be good

**Review by nfalco79 — APPROVED** [review]

**Pull request merged**